gnd/src/main/java/com/google/android/gnd/ui/editobservation/EditObservationFragment.java

```diff
@@ -133,19 +134,18 @@ private void handleSaveResult(EditObservationViewModel.SaveResult saveResult) {
   }
 
   private void addFieldViewModel(Field field, AbstractFieldViewModel fieldViewModel) {
-    fieldViewModel.init(field, viewModel.getResponse(field.getId()));
+    // TODO: Get initial response via parameters.
+    fieldViewModel.init(field, viewModel.getSavedOrOriginalResponse(field.getId()));
 
     if (fieldViewModel instanceof PhotoFieldViewModel) {
       observeSelectPhotoClicks((PhotoFieldViewModel) fieldViewModel);
+      observePhotoAdded((PhotoFieldViewModel) fieldViewModel);
     } else if (fieldViewModel instanceof MultipleChoiceFieldViewModel) {
       observeMultipleChoiceClicks((MultipleChoiceFieldViewModel) fieldViewModel);
     }
 
-    fieldViewModel
-        .getResponse()
-        .observe(this, response -> viewModel.onResponseChanged(field, response));
-
-    fieldViewModel.getError().observe(this, error -> viewModel.onErrorChanged(field, error));
+    fieldViewModel.getResponse().observe(this, r -> viewModel.onResponseChanged(field, r));
+    fieldViewModel.getError().observe(this, e -> viewModel.onErrorChanged(field, e));
```

Comment:
Do we need to sync the state of each field view model with the state of the edit dialog view model? In general duplicate state is best avoided - it increases complexity, can lead to race conditions, and can be hard to debug. Any way to consolidate this state only in the fields, then query their state when it's needed?